core-bundle/src/DependencyInjection/Compiler/AddAssetsPackagesPass.php

```diff
@@ -70,6 +71,13 @@ private function addBundles(ContainerBuilder $container): void
 
                 $container->setDefinition('assets._version_'.$packageName, $def);
                 $packageVersion = new Reference('assets._version_'.$packageName);
+            } elseif ('contao_core' === $packageName) {
```

Comment:
Could be built as a general solution with a new interface: `if ($bundle instance of VersionProvidingBundleInterface)`

Then everybody could benefit from it by implementing the interface.

Also, this approach would leak the version of Contao again (we did hide it everywhere in the back end etc.).
So I think it should something along the lines of `substr(sha1($secret . $version), 0, 8)`.